eclass/zig-utils.eclass

```diff
@@ -0,0 +1,527 @@
+				esac
+			fi
+			;;
+		riscv32 | riscv64)
+			local c_cpu="${c_flags_mcpu}"
+			case "${c_cpu}" in
+				"")
+					case "${c_arch}" in
+						riscv32) base_cpu=generic_rv32;;
+						riscv64) base_cpu=generic_rv64;;
+					esac
+					;;
+				*) base_cpu="${c_cpu//[-.]/_}";;
+			esac
+
+			local base_isa="${c_flags_march:0:4}"
+			local extensions="${c_flags_march:4}"
+
+			case "${base_isa}" in
+				"" | rv32 | rv64) ;;
+				*) die -n "Unknown RISC-V architecture: ${base_isa}";;
+			esac
+
+			local extension
+			while read -n 1 extension; do
+				case "${extension}" in
+					"") ;;
+					g) features+="+i+m+a+f+d+zicsr+zifencei";;
+					_) die -n "Can't translate multi-letter RISC-V extensions yet";;
+					*) features+="+${extension}";;
+				esac
+			done <<< "${extensions}"
+
+			case "${c_flags_mabi}" in
+				ilp32d | lp64d) features+="+d";;
+				ilp32e | lp64e) features+="+e";;
+				ilp32f | lp64f) features+="+f";;
+				"" | ilp32 | lp64) ;;
+				*) die -n "Unknown RISC-V ABI: ${c_flags_mabi}";;
+			esac
+			;;
+		loongarch64)
+			local c_cpu="${c_flags_march}"
+			case "${c_cpu}" in
+				"") base_cpu=generic_la64;;
+				*) base_cpu="${c_cpu//[-.]/_}";;
+			esac
+
+			case "${c_flags_mabi}" in
+				lp64d) features+="+d";;
+				lp64f) features+="+f";;
+				lp64s | "") ;;
+				*) die -n "Unknown LoongArch ABI: ${c_flags_mabi}";;
+			esac
+			;;
+		powerpc | powerpcle | powerpc64 | powerpc64le)
+			local c_cpu="${c_flags_mcpu}"
+			case "${c_cpu}" in
+				"")
+					case "${c_arch}" in
+						powerpc | powerpcle) base_cpu=ppc;;
+						powerpc64 | powerpc64le) base_cpu=ppc64;;
+					esac
+					;;
+				G*) base_cpu="${c_cpu//G/g}";;
+				powerpcle) base_cpu=ppc;;
+				powerpc*) base_cpu="${c_cpu//powerpc/ppc}";;
+				power*) base_cpu="${c_cpu//power/pwr}";;
+				*) base_cpu="${c_cpu//[-.]/_}";;
+			esac
+			;;
+		*) base_cpu=generic;;
+	esac
+
+	echo "${base_cpu}${features}"
+}
+
+# @FUNCTION: zig-utils_find_installation
+# @DESCRIPTION:
+# Detects suitable Zig installation and sets ZIG_VER and ZIG_EXE
+# variables.
+#
+# See ZIG_EXE and ZIG_VER descriptions for more information.
+zig-utils_find_installation() {
+	# Adapted from https://github.com/gentoo/gentoo/pull/28986
+	# Many thanks to Florian Schmaus (Flowdalic)!
+
+	[[ -n "${ZIG_SLOT}" ]] || die "${FUNCNAME[0]}: ZIG_SLOT must be set"
+	if ver_test "${ZIG_SLOT}" -lt "0.13"; then
+		die "${ECLASS}: ZIG_SLOT must be >= 0.13, found ${ZIG_SLOT}"
+	fi
+
+	einfo "Searching Zig ${ZIG_SLOT}..."
+
+	local zig_supported_versions=(
+		"9999"
+		"0.13.1"
+		"0.13.0"
+	)
+
+	local base_path="${BROOT}/usr/bin"
+
+	local selected_path selected_ver
+	for selected_ver in "${zig_supported_versions[@]}"; do
+		# Check if candidate satisfies ZIG_SLOT condition.
+		if [[ "${selected_ver}" != "${ZIG_SLOT}"* ]]; then
+			continue
+		fi
+
+		# Prefer "dev-lang/zig" over "dev-lang/zig-bin"
+		local candidate_path
+		for candidate_path in "${base_path}/"zig{,-bin}"-${selected_ver}"; do
```

Comment:
```suggestion
		for candidate_path in "${base_path}"/zig{,-bin}-"${selected_ver}"; do
```

Again, not obligatory but IMO it's more readable when separators don't go inside quotes.

Reply:
1d9c23b3a9cc68c38d47c708f1a5861ab576ca51